src/Common/NetworkPerspective.Sync.Application/Domain/Employees/Employee.cs

```diff
@@ -57,14 +57,14 @@ private Employee(EmployeeId id, IEnumerable<Group> groups, bool isExternal, bool
             }
         }
 
-        public static Employee CreateInternal(EmployeeId id, IEnumerable<Group> groups, IDictionary<string, object> props = null, RelationsCollection relations = null)
-            => new Employee(id, groups, false, false, false, props ?? new Dictionary<string, object>(), relations ?? RelationsCollection.Empty);
+        public static Employee CreateInternal(EmployeeId id, IEnumerable<Group> groups, IDictionary<string, object> props = null, RelationsCollection relations = null, IEnumerable<string> groupAccess = null)
+            => new Employee(id, groups, false, false, false, props ?? new Dictionary<string, object>(), relations ?? RelationsCollection.Empty, groupAccess ?? Enumerable.Empty<string>());
 
         public static Employee CreateExternal(string email)
-            => new Employee(EmployeeId.Create(email, string.Empty), Array.Empty<Group>(), true, false, false, ImmutableDictionary<string, object>.Empty, RelationsCollection.Empty);
+            => new Employee(EmployeeId.Create(email, string.Empty), Array.Empty<Group>(), true, false, false, ImmutableDictionary<string, object>.Empty, RelationsCollection.Empty, Enumerable.Empty<string>());
 
         public static Employee CreateBot(string email)
```

Comment:
To na marginesie bo nie tego dotyczy ten PR. Ale tutaj trochę tak jakby bot był pracownikiem (CreateBot zwraca Employee). Tak samo "External" czyli np. klient firmy. Symptomy że ta abstrakcja "Employee" może być nie do końca trafiona, a może nie powinniśmy w ogóle tworzyć tych botów / externali bo i tak koniec końców te informacje nie są przekazywane nigdzie dalej?

Reply:
Pełna zgoda. Myslę że przy okazji możemy usunąć właściwość IsBot.